packages/shared/src/utils/get-filter-operators-for-type.ts

```diff
@@ -62,12 +62,13 @@ export function getFilterOperatorsForType(
 			return ['eq', 'neq', 'lt', 'lte', 'gt', 'gte', 'between', 'nbetween', 'null', 'nnull', 'in', 'nin'];
 
 		case 'geometry':
-			return ['null', 'nnull', 'intersects', 'nintersects', 'intersects_bbox', 'nintersects_bbox'];
+			return ['eq', 'neq', 'null', 'nnull', 'intersects', 'nintersects', 'intersects_bbox', 'nintersects_bbox'];
 
 		default:
 			return [
 				'contains',
 				'ncontains',
+				'icontains',
```

Comment:
```suggestion
```
I believe `icontains` should only be a `text` type filter.

The `contains` in the default clause looks to be serving as a placeholder when a new filter node is added. Ref: https://github.com/directus/directus/pull/9103

On a side note, there are other string operators such as `nicontains`, `istarts_with`, that are in `apply-query.ts`, but not currently usable. I think these should be updated separately, in order to not delay this PR.  https://github.com/directus/directus/blob/c409ffb4c01120663095acc850154a42c3f5dc00/api/src/utils/apply-query.ts#L563-L569

Reply:
yeah, there are probably some others we can go through and make sure are all added such as istarts_with, etc and yes, can be a seperate PR, this one was to make sure we had better consistency to start with and I can do a follow up PR with any other missing ones

Reply:
@licitdev your change broke the unit tests so will fix those again.

Reply:
tests should be fixed again.